omod/src/main/java/org/openmrs/module/bedmanagement/rest/resource/AdmissionLocationResource.java

```diff
@@ -119,14 +150,80 @@ private Object getCustomRepresentationForPatient(BedLayout bedLayout) {
 
     @Override
     public AdmissionLocation getByUniqueId(String uuid) {
-        BedManagementService bedManagementService = (BedManagementService) Context.getModuleOpenmrsServices(BedManagementService.class.getName()).get(0);
-        LocationService locationService = Context.getLocationService();
-        return bedManagementService.getLayoutForWard(locationService.getLocationByUuid(uuid));
+        Location location = Context.getService(LocationService.class).getLocationByUuid(uuid);
+        return Context.getService(BedManagementService.class).getAdmissionLocationByLocation(location);
+    }
+
+    @Override
+    public Object create(SimpleObject propertiesToCreate, RequestContext context) throws ResponseException {
+        if (propertiesToCreate.get("name") == null)
+            throw new ConversionException("The name property is missing");
+
+        Location location = new Location();
```

Comment:
Why do you set the `Location` instance to the `AdmissionLocation` before re-accessing it via `.getWard()`? Why not set all what needs to be set to the `Location` instance first and then set it to `AdmissionLocation` via `setWard(..)`?

Is it possible to turn an existing location into an admission location? The code seems to suggest that one can only create a new location altogether?

Reply:
No, This API for creating new admission location only. So existing location cannot turn to admission location with this API.